packages/paima-sdk/paima-mw-core/src/wallets/evm/injected.ts

```diff
@@ -7,26 +7,40 @@ import { connectInjected } from '../wallet-modes.js';
 import { WalletMode } from '@paima/providers';
 import type { ApiForMode, IProvider } from '@paima/providers';
 import { EvmInjectedConnector } from '@paima/providers';
-import { GlobalConfig } from '@paima/utils';
+import { ConfigNetworkType, GlobalConfig } from '@paima/utils';
 
 interface SwitchError {
   code: number;
 }
 
-async function switchChain(): Promise<boolean> {
-  const mainEvmConfig = await GlobalConfig.mainEvmConfig();
+async function switchChain(chainName: string | undefined): Promise<boolean> {
+  let config = undefined;
+  if (chainName) {
+    const allConfigs = await GlobalConfig.getInstance();
 
-  if (!mainEvmConfig) {
-    // TODO: not sure what this does and what to do here.
-    throw new Error('No evm configuration in switchChain');
-  }
+    config = allConfigs[chainName];
+
+    if (!config) {
+      throw new Error('Configuration for network not found.');
+    }
+    if (config.type !== ConfigNetworkType.EVM && config.type !== ConfigNetworkType.EVM_OTHER) {
+      throw new Error('Configuration is not for an EVM based network.');
+    }
+  } else {
+    const evmConfig = await GlobalConfig.mainEvmConfig();
 
-  const [chainName, config] = mainEvmConfig;
+    if (!evmConfig) {
+      return true;
+    }
+
+    chainName = evmConfig[0];
+    config = evmConfig[1];
```

Comment:
nit: in my opinion, using [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) is usually cleaner and less error-prone than defining a `let` and remembering to set it in all branches